sequence/train/lstm.py

```diff
@@ -0,0 +1,81 @@
+from sequence.model.lstm import det_loss
+from sequence.utils import backward
+from sequence.callbacks import apply as apply_callbacks
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def run_epoch(
```

Comment:
This code is probably very similar to the existing `run_epoch` functions. Maybe you could generalize and reuse some snippets/ maybe not.